src/Microsoft.Azure.WebJobs.Host/Settings/SettingManager.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Azure.WebJobs.Host.Indexers;
+
+namespace Microsoft.Azure.WebJobs.Host.Settings
+{
+    internal class SettingManager
```

Comment:
Instead of adding a new `SettingManager` (we already have a `SettingsManager`), I wonder if we can use an `IConfiguration` in order to handle function metadata (we could populate from `function.json`, environment, and command line via an `IConfigurationBuilder`. @fabiocav might have some more thoughts on this